fabric/tasks.py

```diff
@@ -411,14 +411,12 @@ def execute(task, *args, **kwargs):
             # Otherwise, pull in results from the child run.
             ran_jobs = jobs.run()
             for name, d in six.iteritems(ran_jobs):
-                if d['exit_code'] is not None and d['exit_code'] != 0:
+                if isinstance(d["results"], BaseException):
```

Comment:
I'm not sure it's a great idea to change how the entire library handles errors just to cover one case in `streamparse` ? Can we change this to:

```python
if isinstance(d["results"], BaseException or (d['exit_code'] is not None and d['exit_code'] != 0):

```